### PR TITLE
Forward port #603 to main

### PR DESCRIPTION
**Pull request by mjcarroll**
Forward port #603

### CODE REVIEW

**Review by ahcorde — APPROVED** [review]

File: include/gz/math/Helpers.hh [hunk 1]
```diff
@@ -83,620 +83,627 @@ constexpr T GZ_MASSMATRIX3_DEFAULT_TOLERANCE = T(10);
...
+  constexpr auto GZ_DEPRECATED(7) IGN_ZERO_SIZE_T  = &GZ_ZERO_SIZE_T;
+  constexpr auto GZ_DEPRECATED(7) IGN_ONE_SIZE_T   = &GZ_ONE_SIZE_T;
+  constexpr auto GZ_DEPRECATED(7) IGN_TWO_SIZE_T   = &GZ_TWO_SIZE_T;
+  constexpr auto GZ_DEPRECATED(7) IGN_THREE_SIZE_T = &GZ_THREE_SIZE_T;
+  constexpr auto GZ_DEPRECATED(7) IGN_FOUR_SIZE_T  = &GZ_FOUR_SIZE_T;
+  constexpr auto GZ_DEPRECATED(7) IGN_FIVE_SIZE_T  = &GZ_FIVE_SIZE_T;
+  constexpr auto GZ_DEPRECATED(7) IGN_SIX_SIZE_T   = &GZ_SIX_SIZE_T;
+  constexpr auto GZ_DEPRECATED(7) IGN_SEVEN_SIZE_T = &GZ_SEVEN_SIZE_T;
+  constexpr auto GZ_DEPRECATED(7) IGN_EIGHT_SIZE_T = &GZ_EIGHT_SIZE_T;
+  constexpr auto GZ_DEPRECATED(7) IGN_NINE_SIZE_T  = &GZ_NINE_SIZE_T;
```
Comment (scpeters) [inline include/gz/math/Helpers.hh:131]: these should not be forward-ported

File: include/gz/math/Stopwatch.hh [hunk 2]
```diff
@@ -23,104 +23,102 @@
...
+  /// watch.Stop();
+  /// ```
+  class GZ_MATH_VISIBLE Stopwatch
   {
     // Use a steady clock
     // This alias is now deprecated; please use std::chrono::steady_clock
     // directly instead.
     using clock
         [[deprecated("As of 8.0, use std::chrono::steady_clock directly.")]]
         = std::chrono::steady_clock;
```
Comment (scpeters) [inline include/gz/math/Stopwatch.hh:61]: this deprecated using statement should not be in the `Stopwatch` class

File: include/gz/math/Stopwatch.hh [hunk 3]
```diff
@@ -23,104 +23,102 @@
 #include <gz/math/config.hh>
 #include <gz/utils/ImplPtr.hh>
 
-namespace gz
+namespace gz::math
 {
-  namespace math
+  // Use a steady clock
+  using clock = std::chrono::steady_clock;
```
Comment (scpeters) [inline include/gz/math/Stopwatch.hh:29]: this is old

**Review by scpeters — CHANGES_REQUESTED** [review]

**Comment by scpeters** [comment]
this has conflicts; I've opened https://github.com/gazebosim/gz-math/pull/611 as an alternative

**Comment by mjcarroll** [comment]
Replaced by #611

**Pull request closed without merging**